src/state/mod.rs

```diff
@@ -238,7 +239,10 @@ pub fn apply_block_batch<'a>(
         .map(|x| x.0)
         .unwrap();
 
-    let to_finalize = lastest_immutable_slot(tip, byron, shelley);
+    let to_finalize = max_slots_before_prune
+        .map(|x| tip - x)
```

Comment:
@scarmuega I'm not sure if you want extra guardrails here in case the user shoots themselves in the foot by setting `max_slots_before_prune` to a very small number.